src/test/java/io/odh/test/e2e/standard/PipelineV2ServerST.java

```diff
@@ -0,0 +1,401 @@
+                            .withDbConfigConMaxLifetimeSec(120L)
+                            .withDeploy(true)
+                            .withEnableOauth(true)
+                            .withEnableSamplePipeline(false)
+                            .withInjectDefaultScript(true)
+                            .withStripEOF(true)
+                            .withTerminateStatus(ApiServer.TerminateStatus.CANCELLED)
+                            .withTrackArtifacts(true)
+                        .endApiServer()
+                        .withNewDatabase()
+                            .withDisableHealthCheck(false)
+                            .withNewMariaDB()
+                                .withDeploy(true)
+                                .withPipelineDBName("mlpipeline")
+                                .withNewPvcSize("10Gi")
+                                .withUsername("mlpipeline")
+                            .endMariaDB()
+                        .endDatabase()
+                        .withNewMlmd()
+                            .withDeploy(true)
+                        .endMlmd()
+                        .withNewObjectStorage()
+                            .withDisableHealthCheck(false)
+                            .withNewMinio()
+                                .withDeploy(true)
+                                .withImage("quay.io/minio/minio")
+                                .withNewPvcSize("1Gi")
+                                .withBucket("HolyGrail")
+                                .withNewS3CredentialsSecret()
+                                    .withAccessKey("AWS_ACCESS_KEY_ID")
+                                    .withSecretKey("AWS_SECRET_ACCESS_KEY")
+                                    .withSecretName(secretName)
+                                .endMinioS3CredentialsSecret()
+                            .endMinio()
+                        .endObjectStorage()
+                        .withNewPersistenceAgent()
+                            .withDeploy(true)
+                            .withNumWorkers(2L)
+                        .endPersistenceAgent()
+                        .withNewScheduledWorkflow()
+                            .withCronScheduleTimezone("UTC")
+                            .withDeploy(true)
+                        .endScheduledWorkflow()
+                    .endSpec()
+                    .build();
+            ResourceManager.getInstance().createResourceWithWait(dspa);
+        });
+
+        Allure.step("Wait for Pipeline API server to come up");
+        Resource<Endpoints> endpoints = client.endpoints().inNamespace(prjTitle).withName("ds-pipeline-pipelines-definition");
+        waitForEndpoints(endpoints);
+
+        Allure.step("Connect to the API server");
+        Resource<Route> route = ocClient.routes()
+                .inNamespace(prjTitle).withName("ds-pipeline-pipelines-definition");
+
+        // TODO(jdanek) I still don't know how to do oauth, so let's forward a port
```

Comment:
Expose service via route (http) ? Or run it inside the cluster itself. Just an idea.

Reply:
I hope I'll get the OAuth working eventually. At the latest when https://issues.redhat.com/browse/RHOAIENG-2822 is resolved.

Reply:
ok, it was just a suggestion. I can live with current way